third_party/github.com/docker/libcompose/project/project.go

```diff
@@ -31,15 +33,30 @@ type Event struct {
 }
 
 // NewProject create a new project with the specified context.
-func NewProject(context *Context) *Project {
+func NewProject(context *Context) (*Project, error) {
 	p := &Project{
 		context: context,
 		Configs: make(map[string]*ServiceConfig),
 	}
 
+	if context.EnvironmentLookup == nil {
```

Comment:
Is this in the upstream lib compose too?  You could also turn this into a helper method (to create the env lookup) and then just call that from generate (without having to change this method).